checks.d/mapreduce.py

```diff
@@ -0,0 +1,487 @@
+    'mapsCompleted': ('mapreduce.job.maps_completed', GAUGE),
+    'reducesTotal': ('mapreduce.job.reduces_total', GAUGE),
+    'reducesCompleted': ('mapreduce.job.reduces_completed', GAUGE),
+    'mapsPending': ('mapreduce.job.maps_pending', GAUGE),
+    'mapsRunning': ('mapreduce.job.maps_running', GAUGE),
+    'reducesPending': ('mapreduce.job.reduces_pending', GAUGE),
+    'reducesRunning': ('mapreduce.job.reduces_running', GAUGE),
+    'newReduceAttempts': ('mapreduce.job.new_reduce_attempts', GAUGE),
+    'runningReduceAttempts': ('mapreduce.job.running_reduce_attempts', GAUGE),
+    'failedReduceAttempts': ('mapreduce.job.failed_reduce_attempts', GAUGE),
+    'killedReduceAttempts': ('mapreduce.job.killed_reduce_attempts', GAUGE),
+    'successfulReduceAttempts': ('mapreduce.job.successful_reduce_attempts', GAUGE),
+    'newMapAttempts': ('mapreduce.job.new_map_attempts', GAUGE),
+    'runningMapAttempts': ('mapreduce.job.running_map_attempts', GAUGE),
+    'failedMapAttempts': ('mapreduce.job.failed_map_attempts', GAUGE),
+    'killedMapAttempts': ('mapreduce.job.killed_map_attempts', GAUGE),
+    'successfulMapAttempts': ('mapreduce.job.successful_map_attempts', GAUGE),
+}
+
+MAPREDUCE_JOB_COUNTER_METRICS = {
+    'reduceCounterValue': ('mapreduce.job.counter.reduce_counter_value', GAUGE),
+    'mapCounterValue': ('mapreduce.job.counter.map_counter_value', GAUGE),
+    'totalCounterValue': ('mapreduce.job.counter.total_counter_value', GAUGE),
+}
+
+MAPREDUCE_TASK_METRICS = {
+    'progress': ('mapreduce.job.task.progress', GAUGE),
+    'elapsedTime': ('mapreduce.job.task.elapsed_time', GAUGE)
+}
+
+MAPREDUCE_TASK_STATE_METRICS = {
+    'NEW': ('maprduce.job.task.state.new', GAUGE),
+    'SCHEDULED': ('maprduce.job.task.state.scheduled', GAUGE),
+    'RUNNING': ('maprduce.job.task.state.running', GAUGE),
+    'SUCCEEDED': ('maprduce.job.task.state.succeeded', GAUGE),
+    'FAILED': ('maprduce.job.task.state.failed', GAUGE),
+    'KILL_WAIT': ('maprduce.job.task.state.killWait', GAUGE),
+    'KILLED': ('maprduce.job.task.state.killed', GAUGE),
+}
+
+MAPREDUCE_TASK_TYPE_METRICS = {
+    'MAP': ('maprduce.job.task.type.map', GAUGE),
+    'REDUCE': ('maprduce.job.task.type.reduce', GAUGE),
+}
+
+MAPREDUCE_TASK_COUNTER_METRICS = {
+    'MAP_INPUT_RECORDS': ('mapreduce.job.task.counter.map_input_records', GAUGE),
+    'MAP_OUTPUT_RECORDS': ('mapreduce.job.task.counter.map_output_records', GAUGE),
+    'COMBINE_INPUT_RECORDS': ('mapreduce.job.task.counter.combine_input_records', GAUGE),
+    'COMBINE_OUTPUT_RECORDS': ('mapreduce.job.task.counter.combine_output_records', GAUGE),
+    'REDUCE_INPUT_RECORDS': ('mapreduce.job.task.counter.reduce_input_records', GAUGE),
+    'REDUCE_OUTPUT_RECORDS': ('mapreduce.job.task.counter.reduce_output_records', GAUGE),
+}
+
+class MapReduceCheck(AgentCheck):
+
+    def check(self, instance):
+        # Get properties from conf file
+        rm_address = instance.get('resourcemanager_uri', DEFAULT_RM_URI)
```

Comment:
I'd say we should require `resourcemanager_uri` as it's the only parameter of the instance configuration, and we require instances to have at least one parameter in the yaml config. Also, the value in the example yaml file makes it clear what the default URI is.